README.md

```diff
@@ -1329,6 +1332,68 @@ A Map usage example is shown below.
     map.replace("key", "value", "newvalue");
 ```
 
+#### 7.4.1.1. Using IMap Non-Blocking Async Methods
+Hazelcast IMap provides asynchronous methods for more powerful operations like batched writing or batched reading. Asynchronous methods do not block but return immediately with an `ICompletableFuture` interface. You can later query the result of the operation using this interface or you can even provide a callback method to be executed on the user executor threads when the response to the call is received. 
+
+An `Imap::putAsync` usage example is shown below:
+
+```C++
+    // initiate map put in an unblocking way
+    boost::shared_ptr<ICompletableFuture<std::string> > future = map.putAsync("key", "value");
+
+    // do some other work
+    // ----
+    
+    // later on get the result of the put operation
+    boost::shared_ptr<std::string> result = future->get();
+    if (result.get()) {
+        std::cout << "There was a previous value for key. The value was:" << *result << std::endl;
+    } else {
+        std::cout << "There was no previous value for key." << std::endl;
+    }
+```
+The asynchronous methods always return an `ICompletableFuture` pointer. The method does not block for IO. You can get the result later with the `ICompletableFuture::get()` method. `ICompletableFuture::get()` blocks until the response is received or an exception occurs.  
+
+Other asynchronous `IMap` methods are:
+ 
+- `IMap::getAsync`
+- `IMap::removeAsync`
+- `IMap::setAsync`
+- `IMap::getAsync`
+
+You can also provide a callback instance to the future using the `andThen` method. The callback is executed upon reception of the call response or exception and you can put your code into the callback methods. The callback method will be executed in the user executor thread which is different from the user's main program thread and hence it will not block the user. An example callback usage is shown below:
```

Comment:
... executed upon the reception of the call response or exception, and you can ...